app/assets/javascripts/components/option-select.js

```diff
@@ -180,53 +196,58 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
+      this.$optionSelect.querySelector('.js-container-button').setAttribute('aria-expanded', true)
+      this.$optionSelect.classList.remove('js-closed')
+      this.$optionSelect.classList.add('js-opened')
+      if (!this.$optionsContainer.style.height) {
         this.setupHeight()
       }
     }
   }
 
   OptionSelect.prototype.close = function close () {
-    this.$optionSelect.removeClass('js-opened')
-    this.$optionSelect.addClass('js-closed')
-    this.$optionSelect.find('.js-container-button').attr('aria-expanded', false)
+    this.$optionSelect.classList.remove('js-opened')
+    this.$optionSelect.classList.add('js-closed')
+    this.$optionSelect.querySelector('.js-container-button').setAttribute('aria-expanded', false)
   }
 
   OptionSelect.prototype.isClosed = function isClosed () {
-    return this.$optionSelect.hasClass('js-closed')
+    return this.$optionSelect.classList.contains('js-closed')
   }
 
   OptionSelect.prototype.setContainerHeight = function setContainerHeight (height) {
-    this.$optionsContainer.css({
-      height: height
-    })
+    this.$optionsContainer.setAttribute('style', 'height: ' + height + 'px')
   }
 
-  OptionSelect.prototype.isCheckboxVisible = function isCheckboxVisible (index, option) {
-    var $checkbox = $(option)
-    var initialOptionContainerHeight = this.$optionsContainer.height()
-    var optionListOffsetTop = this.$optionList.offset().top
-    var distanceFromTopOfContainer = $checkbox.offset().top - optionListOffsetTop
+  OptionSelect.prototype.isCheckboxVisible = function isCheckboxVisible (option) {
+    var initialOptionContainerHeight = this.getHeight(this.$optionsContainer)
+    var optionListOffsetTop = this.$optionList.getBoundingClientRect().top
+    var distanceFromTopOfContainer = option.getBoundingClientRect().top - optionListOffsetTop
     return distanceFromTopOfContainer < initialOptionContainerHeight
   }
 
   OptionSelect.prototype.getVisibleCheckboxes = function getVisibleCheckboxes () {
-    var visibleCheckboxes = this.$options.filter(this.isCheckboxVisible.bind(this))
+    var visibleCheckboxes = []
+    for (var i = 0; i < this.$options.length; i++) {
+      if (this.isCheckboxVisible(this.$options[i])) {
+        visibleCheckboxes.push(this.$options[i])
+      }
+    }
+
     // add an extra checkbox, if the label of the first is too long it collapses onto itself
-    visibleCheckboxes = visibleCheckboxes.add(this.$options[visibleCheckboxes.length])
+    if (this.$options[visibleCheckboxes.length]) {
+      visibleCheckboxes.push(this.$options[visibleCheckboxes.length])
+    }
     return visibleCheckboxes
   }
 
   OptionSelect.prototype.setupHeight = function setupHeight () {
-    var initialOptionContainerHeight = this.$optionsContainer.height()
-    var height = this.$optionList.outerHeight(true)
+    var initialOptionContainerHeight = this.getHeight(this.$optionsContainer)
+    var height = this.outerHeight(this.$optionList)
```

Comment:
given we only use this function once I'm not sure we need to an add an extra function for it - particularly if we can anticipate `this.$optionList` exists - we can just do `this.$optionList.offsetHeight` and save the context jump for a reader.

Reply:
Function removed and `offsetHeight` added where required.  Thanks